### PR TITLE
Devise token Auth

### PR DESCRIPTION
**Pull request by mpuricawolox**
## Summary

Installing and configuring devise token auth

## Trello Card

https://trello.com/c/qaAu0KxQ/26-implementar-autenticaci%C3%B3n

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -66,3 +65,7 @@ end
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 
 gem "pg", "~> 1.0"
+
+gem "devise_token_auth", "~> 0.2.0"
+
+gem "devise", "~> 4.4"
```
Comment (leamotta) [inline Gemfile:72]: did you have to change the devise version for something in particular?

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@
   # Devise will use the `secret_key_base` as its `secret_key`
   # by default. You can change it below and use your own secret key.
   # config.secret_key = '79c8226c2c21085816562ba8fb57a483613dfcfb86920d25f4609f14f21d285fe60b4ca3e945cf72a61f16a369e19b15cd07fcde3c518c5871a70e120729df68'
-  
+  config.secret_key = '7b27c06d030597a433d1dc5feb5e728eb9d485acdec55438a3a79f9b7ce1cc006682ba7e1c027c4ca09adfcf15576e5388882101c1716b31f55dcce6f7b476a0'
```
Comment (leamotta) [inline config/initializers/devise.rb]: 🤔 why not keep using the secret key base?

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
 Rails.application.routes.draw do
-  devise_for :users
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :listings
```
Comment (leamotta) [inline config/routes.rb]: what's listings?

File: db/migrate/20180828174803_devise_token_auth_create_users.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
+class DeviseTokenAuthCreateUsers < ActiveRecord::Migration[5.2]
+  def change  
+
+    create_table(:users) do |t|
```
Comment (leamotta) [inline db/migrate/20180828174803_devise_token_auth_create_users.rb]: this doesn't work if you already have the migration run in a production or testing environment. You can't just do a drop of the database and a new generate.
What we suggest doing is: keep the old migrations as they were and create a new migration adding just the fields required by devise token auth. That way it's safer.
Keep this in mind for the future, it's very rare to change a migration already run in the master branch.

**Review by leamotta — CHANGES_REQUESTED** [review]
a few comments and things to know

File: config/initializers/devise.rb (hunk 2 above)
Comment (leamotta) [reply config/initializers/devise.rb]: this way you expose to everyone that can access the codebase the secret key that devise uses. It's safer to use an environment variable like the secret key

File: Gemfile (hunk 1 above)
Comment (mpuricawolox) [reply Gemfile:72]: Yeah devise token auth didn't like the other one... So I uninstall them both, and reinstall first token auth and then devise...

File: config/routes.rb (hunk 3 above)
Comment (mpuricawolox) [reply config/routes.rb]: My bad. It was an entity I created for testing.

File: db/migrate/20180828174803_devise_token_auth_create_users.rb (hunk 4 above)
Comment (mpuricawolox) [reply db/migrate/20180828174803_devise_token_auth_create_users.rb]: Ok, greate.

File: config/initializers/devise.rb (hunk 2 above)
Comment (mpuricawolox) [reply config/initializers/devise.rb]: So what would be the best aproach? Creating the env variable directly in the OS (ie: export my_env_var="secret") and just capturing it in my code like ENV['my_env_var'] or using configuration files containing the env (adding to .gitignore for production)?

**Review by leamotta — APPROVED** [review]
Congratz!

**Pull request merged**